Client/Extensions/UIImageViewExtensions.swift

```diff
@@ -9,46 +9,55 @@ import Shared
 
 public extension UIImageView {
 
-    public func setIcon(_ icon: Favicon?, forURL url: URL?, completed completionBlock: ((UIColor, URL?) -> Void)? = nil ) {
+    public func setIcon(_ icon: Favicon?, forURL url: URL?, completed completion: ((UIColor, URL?) -> Void)? = nil ) {
+        func finish(filePath: String?, bgColor: UIColor) {
+            if let filePath = filePath {
+                self.image = UIImage(contentsOfFile: filePath)
+            }
+            // If the background color is clear, we may decide to set our own background based on the theme.
+            let color = bgColor.components.alpha < 0.01 ? UIColor.theme.general.faviconBackground : bgColor
```

Comment:
This is the fix, the rest is just refactoring to ensure (and clarify) that the `finish` block is always hit.